eth2util/eth2exp/attagg.go

```diff
@@ -0,0 +1,151 @@
+}
+
+// MarshalJSON implements json.Marshaler.
+func (b *BeaconCommitteeSubscription) MarshalJSON() ([]byte, error) {
+	resp, err := json.Marshal(&beaconCommitteeSubscriptionJSON{
+		ValidatorIndex:   fmt.Sprintf("%d", b.ValidatorIndex),
+		Slot:             fmt.Sprintf("%d", b.Slot),
+		CommitteeIndex:   fmt.Sprintf("%d", b.CommitteeIndex),
+		CommitteesAtSlot: fmt.Sprintf("%d", b.CommitteesAtSlot),
+		SlotSignature:    fmt.Sprintf("%#x", b.SlotSignature),
+	})
+	if err != nil {
+		return nil, errors.Wrap(err, "marshal beacon committee subscriptions v2")
+	}
+
+	return resp, nil
+}
+
+// UnmarshalJSON implements json.Unmarshaler.
+func (b *BeaconCommitteeSubscription) UnmarshalJSON(input []byte) error {
+	var err error
+
+	var beaconCommitteeSubscriptionJSON beaconCommitteeSubscriptionJSON
+	if err = json.Unmarshal(input, &beaconCommitteeSubscriptionJSON); err != nil {
+		return errors.Wrap(err, "invalid JSON")
+	}
+	if beaconCommitteeSubscriptionJSON.ValidatorIndex == "" {
+		return errors.New("validator index missing")
+	}
+	validatorIndex, err := strconv.ParseUint(beaconCommitteeSubscriptionJSON.ValidatorIndex, 10, 64)
+	if err != nil {
+		return errors.Wrap(err, "invalid value for validator index")
+	}
+	b.ValidatorIndex = eth2p0.ValidatorIndex(validatorIndex)
+	if beaconCommitteeSubscriptionJSON.Slot == "" {
+		return errors.New("slot missing")
+	}
+	slot, err := strconv.ParseUint(beaconCommitteeSubscriptionJSON.Slot, 10, 64)
+	if err != nil {
+		return errors.Wrap(err, "invalid value for slot")
+	}
+	b.Slot = eth2p0.Slot(slot)
+	if beaconCommitteeSubscriptionJSON.CommitteeIndex == "" {
+		return errors.New("committee index missing")
+	}
+	committeeIndex, err := strconv.ParseUint(beaconCommitteeSubscriptionJSON.CommitteeIndex, 10, 64)
+	if err != nil {
+		return errors.Wrap(err, "invalid value for committee index")
+	}
+	b.CommitteeIndex = eth2p0.CommitteeIndex(committeeIndex)
+	if beaconCommitteeSubscriptionJSON.CommitteesAtSlot == "" {
+		return errors.New("committees at slot missing")
+	}
+	if b.CommitteesAtSlot, err = strconv.ParseUint(beaconCommitteeSubscriptionJSON.CommitteesAtSlot, 10, 64); err != nil {
+		return errors.Wrap(err, "invalid value for committees at slot")
+	}
+	if b.CommitteesAtSlot == 0 {
+		return errors.New("committees at slot cannot be 0")
+	}
+
+	if beaconCommitteeSubscriptionJSON.SlotSignature == "" {
+		return errors.New("slot signature missing")
+	}
+
+	signature, err := hex.DecodeString(strings.TrimPrefix(beaconCommitteeSubscriptionJSON.SlotSignature, "0x"))
+	if err != nil {
+		return errors.Wrap(err, "invalid value for signature")
+	}
+	if len(signature) != eth2p0.SignatureLength {
+		return errors.New("incorrect length for signature")
+	}
+	copy(b.SlotSignature[:], signature)
+
+	return nil
+}
+
+// String returns a string version of the structure.
+func (b *BeaconCommitteeSubscription) String() string {
+	data, err := json.Marshal(b)
+	if err != nil {
+		return fmt.Sprintf("ERR: %v", err)
```

Comment:
mmm

Reply:
this too took directly from go-eth2-client, will change